src/proj.h

```diff
@@ -0,0 +1,205 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/******************************************************************************
+    proj.h is included by projects.h in order to determine the size of the
+    OBSERVATION object.
+
+    In order to stomp as little as possible on the traditional proj.4 name
+    space, proj.h is littered with inclusion guards, leaving only the minimum
+    possible implementation when included from projects.h
+
+    The OBSERVATION object is fully defined if proj.h is included alone or
+    in connection with *but before* projects.h (the latter may be needed in
+    some cases, where it is necessary to access this "bare essentials" API,
+    while still having direct access to PJ object internals)
+******************************************************************************/
+
+/* Data type for generic geodetic observations */
+struct OBSERVATION;
+typedef struct OBSERVATION OBSERVATION;
+
+/* Data type for generic geodetic 3D data */
+union PJ_TRIPLET;
+typedef union PJ_TRIPLET PJ_TRIPLET;
+
+/* Data type for generic geodetic 3D data plus epoch information */
+union PJ_SPATIOTEMPORAL;
+typedef union PJ_SPATIOTEMPORAL PJ_SPATIOTEMPORAL;
+
+#ifndef PROJECTS_H
+/* Data type for projection/transformation information */
+struct PJconsts;
+typedef struct PJconsts PJ;         /* the PJ object herself */
+
+/* Omega, Phi, Kappa: Rotations */
+typedef struct {double o, p, k;}  OPK;
+
+/* Easting, Northing, and some kind of height (orthometric or ellipsoidal) */
+typedef struct {double e, n, h;}  ENH;
+
+/* Geodetic spatiotemporal coordinate types */
+typedef struct { double   x,   y,  z, t; }  XYZT;
+typedef struct { double   e,   n,  h, t; }  ENHT;
+typedef struct { double   u,   v,  w, t; }  UVWT;
+typedef struct { double lam, phi,  z, t; }  LPZT;
+
+/* Classic proj.4 pair/triplet types */
+typedef struct { double   u,   v; }  UV;
+typedef struct { double   x,   y; }  XY;
+typedef struct { double lam, phi; }  LP;
+
+typedef struct { double   x,   y,  z; }  XYZ;
+typedef struct { double   u,   v,  w; }  UVW;
+typedef struct { double lam, phi,  z; }  LPZ;
+
+/* Ancillary pairs and triplets for geodetic computations */
+
+/* Degrees, minutes, and seconds */
+typedef struct { double d, m,  s; }  DMS;
+
+/* Geoid undulation and deflections of the vertical */
+typedef struct { double   N,   z,  e; }  PJ_ANC_GEOIDAL;
+
+/* Ellipsoidal parameters */
+typedef struct { double   a,   f; }  PJ_ANC_ELLIPSOIDAL;
+#endif  /* ndef PROJECTS_H */
+
+union PJ_SPATIOTEMPORAL {
+#ifndef PROJECTS_H
+    XYZT xyzt;
+    UVWT uvwt;
+    ENHT enht;
+    LPZT lpzt;
+    ENH  enh;
+#endif
+    double v[4]; /* Who cares - it's just a vector! */
+    XYZ  xyz;
+    UVW  uvw;
+    LPZ  lpz;
+    XY   xy;
+    UV   uv;
+    LP   lp;
+};
+
+
+
+/* Avoid preprocessor renaming and implicit type-punning: Use a union to make it explicit */
+union PJ_TRIPLET {
+#ifndef PROJECTS_H
+    OPK    opk;
+    ENH    enh;
+    PJ_ANC_GEOIDAL     nze;
```

Comment:
"N" is the commonly used designation for the geoid undulation. "z" and "e" are latinizations of the greek zeta and eta, which are commonly used designations for the components of the deflection of the vertical.